.github/workflows/review-code.yaml

```diff
@@ -40,17 +40,21 @@ jobs:
       # Step 5: Send PR Diff for Each File Separately
       - name: Send PR Diff for Each File Separately
         run: |
-          diff_content=$(cat diff.txt | jq -Rsa .)
           # Loop through each file path and send it in a separate request
           while IFS= read -r file; do
             echo "Processing file: $file"
-            file_content=$(git show "${{ github.event.pull_request.head.sha }}:$file" | jq -Rsa .)  # Get the full file content
+
+            # Get the specific diff for this file
+            file_diff=$(git diff -U3 origin/${{ github.event.pull_request.base.ref }}...origin/${{ github.event.pull_request.head.ref }} -- "$file" | jq -Rsa .)
+
+            # Get the full content of the file at the current commit
+            file_content=$(git show "${{ github.event.pull_request.head.sha }}:$file" | jq -Rsa .)
 
             # Send the diff and file content to the backend for each file separately
             curl -X POST "https://app5-latest.onrender.com/openai/api/v2/strategy/4/review-pr-diff" \
               -H "Content-Type: application/json" \
               -d '{
-                "diff": '"$diff_content"',
+                "diff": '"$file_diff"',
```

Comment:
The usage of single and double quotes in the 'diff' value assignment may lead to unexpected behavior. Ensure proper quoting for variables. (Severity: low)